src/components/contextual/stake/StakePreview.vue

```diff
@@ -49,9 +49,15 @@ const {
   stakeBPT,
   unstakeBPT,
 } = useStaking();
+
+const numSharesToModify = computed(() =>
```

Comment:
are you sure it should be computed? 🤔

Reply:
may be just
```
const numSharesToModify = props.action === 'stake'
    ? balanceFor(getAddress(props.pool.address))
    : stakedSharesForProvidedPool.value
```

Reply:
Good suggestion 👍

Reply:
What if `stakedSharesForProvidedPool` changes?

Reply:
Probably best to leave it in a computed property if any of the underlying values are reactive. Unless I'm missing something?

Reply:
hm, there is not computed property in code below
```
const shareBalanceToDisplay = ref(
  props.action === 'unstake'
    ? stakedSharesForProvidedPool.value
    : balanceFor(props.pool.address)
);
```

Reply:
I don't understand why you're not referencing this piece of code? We were discussing `numSharesToModify`.  Regardless I wasn't specifically talking about underlying computed properties, but reactive. `props.action` is reactive and `stakedSharesForProvidedPool.value` is reactive in the code snippet you just shared. `props.action` and `stakedSharesForProvidedPool.value` is reactive in the property in question, which means it probably shouldn't be a static variable.

Reply:
as I understand the result of this code is exactly the same as it is in `numSharesToModify`, but written in other way. I mean previously  there was not need in reactive property there. maybe I am wrong, but I think there is not now as 
`numSharesToModify`  is used only on approval stage.